lib/active_admin/async_exporter/reports/worker.rb

```diff
@@ -8,35 +8,48 @@ class Worker
       def perform(options = {})
         controller = options['controller'].classify.constantize.new
         columns = options['columns']
+        query = query(options['query'])
+        decorate = options['decorate_model']
+        filename = filename(controller)
 
-        path = Rails.root.join('tmp', filename(controller))
+        path = Rails.root.join('tmp', filename)
 
         CSV.open(path, 'wb', headers: true) do |csv|
-          build_csv(csv, columns, controller, options)
+          build_csv(csv, columns, controller, query, decorate)
         end
 
-        AdminReport.find(options['admin_report_id']).update_attributes(status: :ready)
+        file = Services::AwsS3Service.new(path: path, name: filename).store
+        AdminReport.find(options['admin_report_id']).update_attributes(
+          status: :ready,
+          location_url: file.url
+        )
       end
 
       private
 
-      def collection(controller, options)
-        controller.current_collection.ransack(options['query']).result
+      def query(query_string)
+        return nil unless query_string.present?
+
+        JSON.parse(query_string.gsub('=>', ':'))
```

Comment:
🤔 why is this? you are trying to convert a stringified ruby hash into a valid json object?